service/qos/service_level_controller.cc

```diff
@@ -244,19 +257,33 @@ void service_level_controller::update_from_distributed_data(std::chrono::duratio
         throw std::runtime_error(format("Service level updates from distributed data can only be activated on shard {}", global_controller));
     }
     if (_global_controller_db->distributed_data_update.available()) {
+        sl_logger.info("update_from_distributed_data: starting configuration polling loop");
         _global_controller_db->distributed_data_update = repeat([this, interval] {
             return sleep_abortable<steady_clock_type>(std::chrono::duration_cast<steady_clock_type::duration>(interval),
                     _global_controller_db->dist_data_update_aborter).then_wrapped([this] (future<>&& f) {
-                    try {
-                        f.get();
-                        return update_service_levels_from_distributed_data().then([this] {
-                                return stop_iteration::no;
-                        });
-                    }
-                    catch (const sleep_aborted& e) {
-                        return make_ready_future<seastar::bool_class<seastar::stop_iteration_tag>>(stop_iteration::yes);
-                    }
-                });
+                try {
+                    f.get();
+                    return update_service_levels_from_distributed_data().then_wrapped([] (future<>&& f){
+                        try {
+                            f.get();
+                        } catch (...) {
+                            sl_logger.warn("update_from_distributed_data: exception occurred in distributed"
+                                    " data check loop: {}", std::current_exception());
+                        }
+                        return stop_iteration::no;
+                    });
+                } catch (const sleep_aborted& e) {
+                    sl_logger.info("update_from_distributed_data: configuration polling loop aborted");
+                    return make_ready_future<seastar::bool_class<seastar::stop_iteration_tag>>(stop_iteration::yes);
+                }
```

Comment:
Could any exception except sleep_aborted happen here?
I thought you wanted to catch any exception and continue the loop anyway?

Reply:
> Could any exception except sleep_aborted happen here?
The shouldn't, however if we have a bug or an uncovered path they could. This is why the message is an error level 
and the message states that it is unexpected. Short answer: I preferred to be safe than sorry :slightly_smiling_face: